experiments/multilingual-fineweb2-hq/download_and_tokenize_fineweb2_hq.py

```diff
@@ -0,0 +1,107 @@
+"""
+Tokenizes the Fineweb2-HQ dataset splits.
+
+This module defines a function that returns tokenization steps for each dataset split available in the
+Fineweb2 dataset.
+"""
+
+import os.path
+
+from levanter.store.cache import CacheOptions
+
+from experiments.llama import llama3_tokenizer
+from marin.download.huggingface.download import DownloadConfig
+from marin.download.huggingface.download_hf import download_hf
+from marin.execution.executor import ExecutorStep, executor_main, output_path_of, this_output_path, versioned
+from marin.processing.tokenize import TokenizeConfig, tokenize
+from marin.processing.tokenize.data_configs import TokenizerStep
+
+FINEWEB2_DATASETS = {
+    "arb_Arab": ["arb_Arab/*.parquet"],
+    "rus_Cyrl": ["rus_Cyrl/*.parquet"],
+    "cmn_Hani": ["cmn_Hani/*.parquet"],
+    "deu_Latn": ["deu_Latn/*.parquet"],
+    "spa_Latn": ["spa_Latn/*.parquet"],
+    "jpn_Jpan": ["jpn_Jpan/*.parquet"],
+    "fra_Latn": ["fra_Latn/*.parquet"],
+    "ita_Latn": ["ita_Latn/*.parquet"],
+    "por_Latn": ["por_Latn/*.parquet"],
+    "pol_Latn": ["pol_Latn/*.parquet"],
+    "nld_Latn": ["nld_Latn/*.parquet"],
+    "ind_Latn": ["ind_Latn/*.parquet"],
+    "tur_Latn": ["tur_Latn/*.parquet"],
+    "ces_Latn": ["ces_Latn/*.parquet"],
+    "fas_Arab": ["fas_Arab/*.parquet"],
+    "hun_Latn": ["hun_Latn/*.parquet"],
+    "swe_Latn": ["swe_Latn/*.parquet"],
+    "ell_Grek": ["ell_Grek/*.parquet"],
+    "dan_Latn": ["dan_Latn/*.parquet"],
+    "vie_Latn": ["vie_Latn/*.parquet"],
+}
+
+FINEWEB2_HQ_MIXTURE_WEIGHTS = {  # From https://huggingface.co/datasets/epfml/FineWeb2-HQ
+    "arb_Arab": 94 / 1024,
+    "rus_Cyrl": 1.2,  # TiB
+    "cmn_Hani": 784 / 1024,  # in GiB
+    "deu_Latn": 618 / 1024,
+    "spa_Latn": 515 / 1024,
+    "jpn_Jpan": 393 / 1024,
+    "fra_Latn": 483 / 1024,
+    "ita_Latn": 269 / 1024,
+    "por_Latn": 222 / 1024,
+    "pol_Latn": 168 / 1024,
+    "nld_Latn": 160 / 1024,
+    "ind_Latn": 125 / 1024,
+    "tur_Latn": 100 / 1024,
+    "ces_Latn": 104 / 1024,
+    "fas_Arab": 69 / 1024,
+    "hun_Latn": 79 / 1024,
+    "swe_Latn": 61 / 1024,
+    "ell_Grek": 84 / 1024,
+    "dan_Latn": 61 / 1024,
+    "vie_Latn": 59 / 1024,
+}
+
+download_fineweb2_step = ExecutorStep(
```

Comment:
ordinarily we use something like `fineweb_raw` or something, naming things the result of the step, rather than the step itself